go/tools/bzltestutil/init.go

```diff
@@ -45,7 +46,16 @@ func init() {
 	if hasSrcDir && hasWorkspace && RunDir != "" {
 		abs := RunDir
 		if !filepath.IsAbs(RunDir) {
-			abs = filepath.Join(testSrcDir, testWorkspace, RunDir)
+			if strings.HasSuffix(testSrcDir, ".runfiles") && strings.HasPrefix(RunDir, "external/") {
```

Comment:
I see, the real issue seems to be that the Starlark logic that generates `RunDir` is (still?) assuming that Bazel runs the test from the exec root rather than the runfiles root. I am investigating this in https://github.com/bazelbuild/rules_go/pull/3414.